### PR TITLE
[Blockstore] do not calculate logical used blocks metric for proxy overlay disks

### PR DESCRIPTION
**Pull request by BarkovBG**
issue https://github.com/ydb-platform/nbs/issues/356

- LogicalUsedBlocks metric uses for overlay disks
- proxy overlay disks are temporary so we needn't calculate this metric

### CODE REVIEW

File: cloud/blockstore/libs/storage/volume/partition_requests.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ bool ToPartitionRequests(
     Y_UNUSED(requests);
     Y_UNUSED(blockRange);
 
-    Y_ABORT_UNLESS(0);
+    ythrow TServiceError(E_NOT_IMPLEMENTED);
```
Comment (SvartMetal) [inline cloud/blockstore/libs/storage/volume/partition_requests.h:60]: `Y_ABORT_UNLESS(0, "Not implemented")`?

File: cloud/blockstore/libs/storage/protos/part.proto [hunk 2]
```diff
@@ -73,6 +73,8 @@ message TPartitionConfig
 
     // Optional tablet id of base volume.
     uint64 BaseDiskTabletId = 17;
+
+    bool IsProxyOverlay = 18;
```
Comment (SvartMetal) [inline cloud/blockstore/libs/storage/protos/part.proto:77]: "ProxyOverlay" is not Blockstore concept.
Maybe it is better to disable logical used blocks calculation when `IsSystem == true` and `!BaseDiskId.empty()`?
Or even when `IsSystem == true`.

And this fix looks like "kostyl", so write comment about that, please

File: cloud/blockstore/libs/storage/service/service_ut_create.cpp [hunk 3]
```diff
@@ -700,6 +700,65 @@ Y_UNIT_TEST_SUITE(TServiceCreateVolumeTest)
         }
     }
 
+    Y_UNIT_TEST(ShouldCreateVolumeWithMultipartitionBaseDisk)
```
Comment (SvartMetal) [inline cloud/blockstore/libs/storage/service/service_ut_create.cpp:703]: ShouldCreateSystemVolumeWithMultipartitionedBaseDisk

File: cloud/blockstore/libs/storage/volume/partition_requests.h (hunk 1 above)
Comment (BarkovBG) [reply cloud/blockstore/libs/storage/volume/partition_requests.h:60]: nbs craches on Y_ABORT_UNLESS but we want to get readable error

so ythrow is better solution here

File: cloud/blockstore/libs/storage/protos/part.proto (hunk 2 above)
Comment (BarkovBG) [reply cloud/blockstore/libs/storage/protos/part.proto:77]: i dont know that proxy overlay disks is system

fixed

File: cloud/blockstore/libs/storage/partition/part_actor_loadstate.cpp [hunk 4]
```diff
@@ -228,7 +228,7 @@ void TPartitionActor::CompleteLoadState(
     Y_ABORT_UNLESS(State->GetGarbageQueue().AddNewBlobs(args.NewBlobs));
     Y_ABORT_UNLESS(State->GetGarbageQueue().AddGarbageBlobs(args.GarbageBlobs));
 
-    if (State->GetBaseDiskId()) {
+    if (State->GetBaseDiskId() && !partitionConfig.GetIsSystem()) {
```
Comment (SvartMetal) [inline cloud/blockstore/libs/storage/partition/part_actor_loadstate.cpp:231]: Need comment about why we disable logical used blocks calculation for "system" disks

Comment (SvartMetal) [reply cloud/blockstore/libs/storage/partition/part_actor_loadstate.cpp:233]: Still is not answered why we disable that for such disks

Comment (SvartMetal) [reply cloud/blockstore/libs/storage/partition/part_actor_loadstate.cpp:233]: And it would be better to explain what "proxy overlay" disks are used for

**Review by SvartMetal — APPROVED** [review]

**Review by yegorskii — APPROVED** [review]

**Pull request merged**